src/Microphysics2M.jl

```diff
@@ -545,12 +544,37 @@ function effective_radius(
     Br =
         (xr == 0) ? 0 :
         ((SF.gamma((νr + 1) / μr) * xr / SF.gamma((νr + 2) / μr))^(-μr))
+    Ac = (N_liq * Bc^(νc + 1)) / SF.gamma(νc + 1)
+    Ar = (μr * N_rai * Br^((νr + 1) / μr)) / SF.gamma((νr + 1) / μr)
+
+    cloud_3moment = (Bc == 0) ? 0 : (6 * Ac * C^3 / (Bc * C)^4)
```

Comment:
Maybe the logic would be cleaner to just have one `if/else` per function. 

You could define another helper function that computes the effective radius for cloud and rain. Then that one would return a zero in case no particles are present. And then it would be called twice to compute the whole thing. I think it would be more readable. You could maybe even use dispatch to have one function defined twice for cloud and rain.

Reply:
In case I'm not explaining properly, I can add that after you are done with the other review comments.

Reply:
I am not sure I understood what you'd want to compute exactly in that helper function (since the computation of the moments is different for cloud and rain). And should I do it for both Radar reflectivity and effective radius?

For now I updated the code following the other comments, but feel free to tell me how to create those functions (or to do it by yourself if you prefer)!